src/renderer/mod.rs

```diff
@@ -0,0 +1,244 @@
+//! The renderer for [`Snippet`]s
+//!
+//! # Example
+//! ```
+//! use annotate_snippets::renderer::Renderer;
+//! use annotate_snippets::snippet::{Annotation, AnnotationType, Slice, Snippet};
+//! let snippet = Snippet {
+//!     title: Some(Annotation {
+//!         label: Some("mismatched types"),
+//!         id: None,
+//!         annotation_type: AnnotationType::Error,
+//!     }),
+//!     footer: vec![],
+//!     slices: vec![
+//!         Slice {
+//!             source: "Foo",
+//!             line_start: 51,
+//!             origin: Some("src/format.rs"),
+//!             fold: false,
+//!             annotations: vec![],
+//!         },
+//!         Slice {
+//!             source: "Faa",
+//!             line_start: 129,
+//!             origin: Some("src/display.rs"),
+//!             fold: false,
+//!             annotations: vec![],
+//!         },
+//!     ],
+//!  };
+//!
+//!  let renderer = Renderer::styled();
+//!  println!("{}", renderer.render(snippet));
+
+pub mod margin;
+
+use crate::display_list::DisplayList;
+use crate::snippet::Snippet;
+use anstyle::{AnsiColor, Effects, Style};
+use margin::Margin;
+use std::fmt::Display;
+
+/// A renderer for [`Snippet`]s
+#[derive(Clone)]
+pub struct Renderer {
+    anonymized_line_numbers: bool,
+    margin: Option<Margin>,
+    stylesheet: StyleSheet,
+}
+
+impl Renderer {
+    /// Render a snippet into a `Display`able object
+    pub fn render<'a>(&'a self, snippet: Snippet<'a>) -> impl Display + 'a {
+        DisplayList::new(
+            snippet,
+            &self.stylesheet,
+            self.anonymized_line_numbers,
+            self.margin,
+        )
+    }
+
+    /// No terminal styling
+    pub const fn plain() -> Self {
+        Self {
+            anonymized_line_numbers: false,
+            margin: None,
+            stylesheet: StyleSheet::const_default(),
+        }
+    }
+
+    /// Default terminal styling
+    pub const fn styled() -> Self {
+        Self {
+            anonymized_line_numbers: false,
+            margin: None,
+            stylesheet: StyleSheet {
```

Comment:
To make it clearer that some of these fields are meant to be the same as `plain`, should we do
```rust
Self {
    stylesheet: StyleSheet { ... },
    ..Self::plain(),
}
```